static/panes/ir-view.js

```diff
@@ -196,14 +208,71 @@ Ir.prototype.onCompilerClose = function (id) {
 };
 
 Ir.prototype.onSettingsChange = function (newSettings) {
+    this.settings = newSettings;
     this.irEditor.updateOptions({
         contextmenu: newSettings.useCustomContextMenu,
         minimap: {
             enabled: newSettings.showMinimap
-        }
+        },
+        fontFamily: newSettings.editorsFFont
     });
 };
 
+Ir.prototype.onMouseMove = function (e) {
+    if (e === null || e.target === null || e.target.position === null) return;
+    if (this.settings.hoverShowSource === true && this.irCode) {
+        this.clearLinkedLines();
+        var hoverCode = this.irCode[e.target.position.lineNumber - 1];
+        if (hoverCode) {
+            // We check that we actually have something to show at this point!
+            var sourceLine =  hoverCode.source && !hoverCode.source.file ? hoverCode.source.line : -1;
+            this.eventHub.emit('editorLinkLine', this._editorid, sourceLine, false);
+            this.eventHub.emit('panesLinkLine', this._compilerid, sourceLine, false, this);
+        }
+    }
+};
+
+Ir.prototype.updateDecorations = function () {
+    this.prevDecorations = this.irEditor.deltaDecorations(
+        this.prevDecorations, _.flatten(_.values(this.decorations)));
+};
+
+Ir.prototype.clearLinkedLines = function () {
+    this.decorations.linkedCode = [];
+    this.updateDecorations();
+};
+
+Ir.prototype.onPanesLinkLine = function (compilerId, lineNumber, revealLine, sender) {
+    if (Number(compilerId) === this._compilerid) {
+        var lineNums = [];
+        _.each(this.irCode, function (irLine, i) {
+            if (irLine.source && irLine.source.file === null && irLine.source.line === lineNumber) {
+                lineNums.push(i + 1);
+            }
+        });
+        if (revealLine && lineNums[0]) this.irEditor.revealLineInCenter(lineNums[0]);
```

Comment:
It's a shame we can't share this code. There's a lot going on here and if other views want to implement this too, we've put a burden on them too. There's some constants that need to agree at least, for consistency (e.g. the `5000` fadeout timout). I'm not sure what JS magic we can do for the best: a base class for both Compilers and Ir seems overkill. Perhaps a helper class that both embed that does some of this?

Mind you, that's probably good for a follow-up PR, there's much more opportunity than the new code to de-duplicate :)